core/templates/components/forms/custom-forms-directives/edit-thumbnail-modal.component.spec.ts

```diff
@@ -0,0 +1,194 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for Edit Thumbnail Modal Component.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+import { EditThumbnailModalComponent } from './edit-thumbnail-modal.component';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { NO_ERRORS_SCHEMA, Pipe, SimpleChanges } from '@angular/core';
+import { importAllAngularServices } from 'tests/unit-test-utils';
+
+@Pipe({name: 'translate'})
+class MockTranslatePipe {
+  transform(value: string): string {
+    return value;
+  }
+}
+
+class MockActiveModal {
+  dismiss(): void {
+    return;
+  }
+
+  close(): void {
+    return;
+  }
+}
+
+class mockImageObject {
+  source = null;
+  onload = null;
+  constructor() {
+    this.onload = () => {
+      return 'Fake onload executed';
+    };
+  }
+  set src(url) {
+    this.onload();
+  }
+}
+
+class mockReaderObject {
+  result = null;
+  onload = null;
+  constructor() {
+    this.onload = () => {
+      return 'Fake onload executed';
+    };
+  }
+  readAsDataURL(file) {
+    this.onload();
+    return 'The file is loaded';
+  }
+}
+
+describe('Edit Thumbnail Modal Component', () => {
+  let component: EditThumbnailModalComponent;
+  let fixture: ComponentFixture<EditThumbnailModalComponent>;
+  let ngbActiveModal: NgbActiveModal;
+  let closeSpy: jasmine.Spy;
+  let dismissSpy: jasmine.Spy;
+  importAllAngularServices();
+
+  beforeEach(fakeAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      declarations: [
+        EditThumbnailModalComponent,
+        MockTranslatePipe
+      ],
+      providers: [
+        {
+          provide: NgbActiveModal,
+          useClass: MockActiveModal
+        }
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(EditThumbnailModalComponent);
+    component = fixture.componentInstance;
+    ngbActiveModal = TestBed.inject(NgbActiveModal);
+    closeSpy = spyOn(ngbActiveModal, 'close').and.callThrough();
+    dismissSpy = spyOn(ngbActiveModal, 'dismiss').and.callThrough();
+  });
+
+  it('should load a image file in onchange event and save it if it\'s a' +
+    ' svg file', () => {
+    // This throws "Argument of type 'mockImageObject' is not assignable to
+    // parameter of type 'HTMLImageElement'.". This is because
+    // 'HTMLImageElement' has around 250 more properties. We have only defined
+    // the properties we need in 'mockImageObject'.
+    // @ts-expect-error
+    spyOn(window, 'Image').and.returnValue(new mockImageObject());
```

Comment:
you can try `(new mockImageObject) as unknown as HTMLImageElement`.

Reply:
This particular test isn't passing, need some help here